tensorflow/lite/tools/optimize/quantize_model_test.cc

```diff
@@ -1112,6 +1112,40 @@ TEST_F(QuantizeCustomOpTest, VerifyMixedQuantization) {
   }
 }
 
+class QuantizeUnpackTest : public QuantizeModelTest {
+protected:
+    QuantizeUnpackTest() {
+      input_model_ = ReadModel(internal::kModelWithUnpack);
+      readonly_model_ = input_model_->GetModel();
+      readonly_model_->UnPackTo(&model_);
+    }
+};
+
+TEST_F(QuantizeUnpackTest, VerifyUnpack) {
+  auto status = QuantizeModel(&builder_, &model_, &error_reporter_);
+
+  ASSERT_EQ(kTfLiteOk, status);
+
+  const auto subgraph = model_.subgraphs[0].get();
+
+  // The model should only have one input and 2 outputs.
+  EXPECT_EQ(subgraph->inputs.size(), 1);
+  EXPECT_EQ(subgraph->outputs.size(), 2);
+
+  // Ensure quantization parameters before and after unpack,
+  // are preserved after quantization for all outputs of
+  // unpack (which would be tensors[1] and tensors[0].
+  EXPECT_FLOAT_EQ(subgraph->tensors[0]->quantization->scale[0],
```

Comment:
Take a look at QuantizeArgMaxTest/QuantizeMultiInputAddWithReshapeTest with how the input and output tensors are retrieved. I think it's better since it's more robust (in getting the right tensors) and clearer (we see we're getting the nth input or output tensor). I'd also create some intermediate variables to reduce redundancy and add clarity. 

I recognize that some of the unit tests here are doing what's done in this PR.